exporters/trace_stackdriver/src/main/java/io/opencensus/trace/exporter/StackdriverExporter.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.trace.exporter;
+
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.common.annotations.VisibleForTesting;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.export.SpanExporter;
+import io.opencensus.trace.export.SpanExporter.Handler;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.concurrent.GuardedBy;
+
+/**
+ * An OpenCensus span exporter implementation which exports data to Stackdriver Trace.
+ *
+ * <p>Example of usage on Google Cloud VMs:
+ *
+ * <pre><code>
+ *   public static void main(String[] args) {
+ *     StackdriverExporter.createAndRegister("MyStackdriverProjectId");
+ *     ... // Do work.
+ *   }
+ * </code></pre>
+ */
+public final class StackdriverExporter {
+  private static final String REGISTER_NAME = StackdriverExporter.class.getName();
+  private static final List<String> STACKDRIVER_TRACE_WRITER_SCOPE =
+      Collections.singletonList("https://www.googleapis.com/auth/trace.append");
+  private static final Object monitor = new Object();
+
+  @GuardedBy("monitor")
+  private static Handler handler = null;
+
+  /**
+   * Creates and registers the Stackdriver Trace exporter to the OpenCensus library.
+   *
+   * @param credentials a credentials used to authenticate API calls.
+   * @param projectId the cloud project id.
+   */
+  public static void createAndRegisterWithCredentials(Credentials credentials, String projectId) {
+    synchronized (monitor) {
+      if (handler == null) {
+        handler = new StackdriverV1ExporterHandler(credentials, projectId);
```

Comment:
Does it make sense to support exporting traces to Stackdriver in more than one way, such as using more than one projectId?  I think this class should either support multiple calls to register, possibly by returning a new StackdriverExporter instance, or it should prevent `createAndRegisterWithCredentials` from being called twice (with a comment in the Javadoc).  `createAndRegisterWithCredentials` shouldn't silently fail when `handler` is already set.

Reply:
I don't think that makes too much sense to support multiple projectIds. I added exceptions and docs.

Good catch.